src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -4,8 +4,16 @@ var React = require("react/addons");
 var AppHealthComponent = require("../components/AppHealthComponent");
 
 var AppComponent = React.createClass({
+  propTypes: {
+    model: React.PropTypes.object.isRequired,
+    router: React.PropTypes.object.isRequired
+  },
+
   onClick: function () {
-    this.props.onClick(this.props.model);
+    this.props.router.navigate(
```

Comment:
And the descendant components don't have to be passed the global dispatcher; they access it via `this.emit` meaning they have no knowledge of higher-level state.